Terraform add sync configuration (#17834)
pingcap/docs: tidb-cloud/terraform-get-tidbcloud-provider.md

`public_key` and `private_key` are the API key's public key and private key. You can also pass them through the environment variables:

```
export TIDBCLOUD_PUBLIC_KEY=${public_key}
export TIDBCLOUD_PRIVATE_KEY=${private_key}
```

Now, you can use the TiDB Cloud Terraform Provider.

## Step 5. Configure TiDB Cloud Terraform Provider with sync configuration

> Terraform provider >= 0.3.0 support an optional configuration `sync`.

Reviewer: ```suggestion
Terraform provider (>= 0.3.0) supports an optional parameter `sync`.
```

> You can create, update or delete resources synchronously by setting `sync` to `true`. Here is an example:

Reviewer: ```suggestion
By setting `sync` to `true`, you can create, update, or delete resources synchronously. Here is an example:
```

```
provider "tidbcloud" {

>   public_key  = "your_public_key"

Reviewer: ```suggestion
  public_key = "your_public_key"
```

  private_key = "your_private_key"

>   sync        = true

Reviewer: ```suggestion
  sync = true
```

}
```

> We encourage the use of sync, but note that `sync` currently only works on the cluster resource. Please contact us if you need this feature on other resources.

Reviewer: ```suggestion
Setting `sync` to `true` is recommended, but note that `sync` currently only works with the cluster resource. If you need synchronous operations for other resources, please [contact TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md).
```
